src/Symfony/Component/Yaml/Parser.php

```diff
@@ -1009,7 +1011,7 @@ private function isStringUnIndentedCollectionItem(): bool
      */
     private function isBlockScalarHeader(): bool
     {
-        return (bool) self::preg_match('~'.self::BLOCK_SCALAR_HEADER_PATTERN.'$~', $this->currentLine);
+        return '' !== $this->currentLine && (bool) self::preg_match('~'.self::BLOCK_SCALAR_HEADER_PATTERN.'$~', $this->currentLine);
```

Comment:
what about adding `&& \in_array($this->currentLine[0], array('|', '>'), true) &&` here too ?

Reply:
I thought about that too, but the regular expression should also match lines like `foo: |`, shouldn't it?

Reply:
hmm indeed. This regex is not anchored at the start. I missed that

Reply:
I just remembered that I indeed tried to add that check and was wondering now why tests didn't fail then. Turns out this method is totally useless with the current code (see #28187).